cmd/publishing-bot/publisher.go

```diff
@@ -244,25 +244,57 @@ func (p *PublisherMunger) publish() error {
 }
 
 // Run constructs the repos and pushes them.
-func (p *PublisherMunger) Run() (string, string, error) {
+func (p *PublisherMunger) Run() (string, map[string]string, error) {
 	buf := bytes.NewBuffer(nil)
 	p.plog = NewPublisherLog(buf)
 
-	hash, err := p.updateSourceRepo()
-	if err != nil {
-		p.plog.Errorf("%v", err)
-		p.plog.Flush()
-		return p.plog.Logs(), hash, err
+	// If --branches is specified, re-run the bot with different set of rules
+	// per each specified branch.
+	sourceBranches := strings.Split(p.config.SourceBranches, ",")
+	if len(sourceBranches) == 0 {
```

Comment:
`len(sourceBranches) == 0` should mean that nothing is done. Defaulting should be outside of the func.